src/macaron/dependency_analyzer/dependency_resolver.py

```diff
@@ -129,6 +130,17 @@ def add_latest_version(
         url_to_artifact: dict[str, set]
             Used to detect artifacts that have similar repos.
         """
+        if item["url"] == "" and item["version"] != "unspecified" and item["group"] and item["name"]:
+            gav = f"{item['group']}:{item['name']}:{item['version']}"
+            urls = find_repo(gav, ["scm.url"])
+            # TODO decide how to handle multiple returned URLs
```

Comment:
Before picking a URL, we need to validate and check the URL, which happens in [find_valid_url](https://github.com/oracle-samples/macaron/blob/1906f88c3d725ef01433301031b1bec0ea7829c6/src/macaron/dependency_analyzer/dependency_resolver.py#L182). 

So you can just use this function to select a URL. Here is an [example](https://github.com/oracle-samples/macaron/blob/1906f88c3d725ef01433301031b1bec0ea7829c6/src/macaron/dependency_analyzer/cyclonedx.py#L194-L196).

I realized that while checking the changes in the expected outputs in `cyclonedx_apache_maven.json` that contained `http://svn.apache.org/repos/asf/commons/proper/jxpath/trunk` which we don't support as an scm.